course/layouts/home.html

```diff
@@ -30,13 +30,13 @@
                   <div class="homepage-content d-flex">
                     <div class="container-fluid pt-0 pb-4">
                       <div class="row px-3 pb-2 justify-content-between">
-                        {{ $shouldCollapseImageDescription := gt (len $courseImageMetadata.Params.image_metadata.caption) 100 }}
+                        {{ $shouldCollapseImageDescription := gt (len $courseImageMetadata.Params.image_metadata.caption) 130 }}
                         <div class="col-lg-4 d-none d-lg-block pl-0">
```

Comment:
Bug finding: Some `caption` also have HTML in them (usually anchor tag) which is rendered by `.RenderString` and not shown to users but it is included in `len` function, hence caption length becomes greater and expand/collapse is shown but the entire caption fits in the available area and this issue occurs. (`len` function is not accurate on `.RenderString` due to whitespaces in between so comparison length value has been increased but I'd love to hear if there's any other more appropriate solution)

@gumaerc Would you like to share some thoughts?

Reply:
You might be able to pass the rendered string to a variable first before assessing the length.  Something like:

```
{{ $renderedCaption := $courseImageMetadata.Params.image_metadata.caption }}
{{ $shouldCollapseImageDescription := gt (len $renderedCaption) 130 }}
```

Reply:
> You might be able to pass the rendered string to a variable first before assessing the length. Something like:

Okay, I'll update the PR.

> This strikes me as something that should maybe be solved in the frontend though anyway, rather than during templating. Is there some kind of JS / CSS solution we can use to respond to containers with a certain class going over a certain height and add expand collapse that way?

Yeah, We can do it with JS as well, but a generic solution will be required which is applicable to any section which requires expand/collapse, but I guess, the current solution will be more efficient than the JS one.

Reply:
Although if it's not possible to get the raw length of what would be the visible text in the browser after rendering and if the frequency of this issue is more then I guess we should consider something else.

btw I also tried something like this: 
`(len caption | .RenderString)`
but it also not accurate as it seemed like that link html has been replaced by whitespaces

Reply:
`.RenderString` tells Hugo to generate HTML from a given Markdown string at build time, so that would probably be even more verbose.  Ultimately trying to determine the height based on character length is difficult because really that's up to the browser.  At render time, CSS could change properties like the font size or line height which would affect the height.  Ultimately, I think the only truly reliable way to do this is with JS client side by measuring the height.